### DISCUSSION TITLE
macro DISALLOW_COPY_AND_ASSIGN

### DISCUSSION THREAD

**Discussion by lanqsh** [idealvin/coost#203] [General]
https://github.com/idealvin/cocoyaxi/blob/a9fc2fa580243e5e4d48048ea13b1db3bab1b4b8/include/co/def.h#L40

是不是可以考虑将
`    T(const T&&) = delete; 
    void operator=(const T&&) = delete;
`
也加到这个宏里面

**Comment by idealvin** [comment]
我还没见过有谁用 const 右值引用的..

**Reply by lanqsh** [reply]
> 我还没见过有谁用 const 右值引用的..

我还是太年轻了..
移动构造函数和移动赋值函数的参数不应该写成const
有看到用到了移动构造函数，是我画蛇添足了

另外，膜拜一下，代码写的真6

**Reply by idealvin** [reply]
感谢支持
